x-pack/plugins/monitoring/public/views/base_controller.js

```diff
@@ -129,11 +153,21 @@ export class MonitoringViewBaseController {
     // needed for chart pages
     this.onBrush = ({ xaxis }) => {
       const { to, from } = xaxis;
+      const fromTime = moment(from);
+      let toTime = moment(to);
+      if (toTime.unix() - fromTime.unix() < 40) {
```

Comment:
I actually meant to leave a comment there with a link: https://github.com/elastic/kibana/issues/36738 (step 4). There is a limit onto how much you can zoom-in until the api will give you network error. From my testing some dataset ranges can be as little as 8 seconds and some can't go pass 30 seconds. I decided to limit it to no less then 40 seconds to be on the safe side. I still don't know the condition or cause, or where/why this limit is implemented

Reply:
This behavior is ungated currently though right? Are we stopping users from zooming in too much (without your changes)?

Reply:
We're not, but rather than creating a separate issue and fixing it at a later time I decided to sneak it in with this feature. It's a rather nasty bug, since once you trigger it it will start showing the 404 error every 10 seconds (or w/e the interval is):
![Screen Shot 2019-06-18 at 4 29 07 PM](https://user-images.githubusercontent.com/27822304/59717778-b499e180-91e6-11e9-9e45-f62327cb9a66.png)
Refreshing won't help since the range state is saved in the URL, the only way to get out of it is by: clicking back, changing the range manually, or clicking on Stack Monitoring again

Reply:
> We're not, but rather than creating a separate issue and fixing it at a later time I decided to sneak it in with this feature. It's a rather nasty bug, ...

It's good that you found this bug and are fixing it, but we should do that in its own PR (no need for a separate issue if you can put up the PR right away). There are a couple of benefits of extracting the bugfix into its own PR:

1. It makes reviewing _this_ PR (and the bugfix PR) easier. By separating concerns and having a PR for each, the changes in each PR become more obvious to reviewers, since each PR then has a more focussed context.

2. Typically, bug fixes will get backported to all active branches whereas features only get backported to the current minor branch (`7.x` today). I realize this PR is categorized as a bugfix PR (personally I think the ability to zoom out is a feature, but that can be debated) so one could argue that the changes related to the zoom out ability + the changes related to the this thread would be backported to all the same branches. However, this won't always be the case so it's best to get into the habit of breaking out individual bugfixes into their own PRs.

3. Smaller PRs generally tend to be easier to backport as there is a smaller chance of conflicts, especially against older branches.

Reply:
Thanks @ycombinator! This makes a lot of sense. I didn't really think about backporting etc.

I will revert this fix, and create a new PR for it